AK/Slugify.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019-2020, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/DeprecatedString.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <cctype>
+
+namespace AK {
+
+inline ErrorOr<String> slugify(String const& _raw, char glue = '-')
+{
+    StringBuilder sb;
+    bool just_processed_space = false;
+
+    for (auto const& ch : _raw.bytes()) {
+        if (std::isalnum(ch) != 0) {
+            sb.append(static_cast<char>(std::tolower(ch)));
+            just_processed_space = false;
+        } else if ((ch == static_cast<unsigned char>(glue) || ch == ' ') && !just_processed_space) {
+            sb.append(' ');
+            just_processed_space = true;
+        }
+    }
+
+    StringView glue_str { &glue, 1 };
+    return TRY(TRY(sb.to_string()).trim_ascii_whitespace())
+        .replace(" "sv, glue_str, ReplaceMode::All);
```

Comment:
Why do we construct a string and then do a replacement on it, instead of inserting the glue character when we see whitespace?

Reply:
It was actually to remove leading and trailing glue. Implementing within loop makes sense.